### PR TITLE
KAR-490 | add service type to next version flow

### PR DESCRIPTION
**Pull request by dkubicki**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 coverage
 node_modules
+.idea
```
Comment (bgalek) [inline .gitignore:3]: this should be in your os :) there is an instruction on askbartosz

File: action.js [hunk 2]
```diff
@@ -1,46 +1,67 @@
 const core = require('@actions/core');
 const { increment } = require('./incrementer');
-const { parse } = require('./tag-parser');
+const { parseLibraryTag, parseServiceTag } = require('./tag-parser');
```
Comment (bgalek) [inline action.js:3]: what do you think about leaving only parse() interface, with additional parameter tagType?

File: action.js [hunk 3]
```diff
@@ -1,46 +1,67 @@
...
+
     const mainBranch = ['master', 'main'].includes(currentBranch);
     core.info(`current branch: ${currentBranch}`)
     core.info(`last tag found ${latestTag}`);
 
     if (!mainBranch) core.info('this is a snapshot release');
     if (latestTag) {
-        const { prefix, version } = parse(latestTag);
+        const { prefix, version } = isLibrary(type) ? parseLibraryTag(latestTag) : parseServiceTag(latestTag)
+        console.log('TUTAJJJJJ PREFIX' + 'TUTAJJJJJ VERSION' + prefix, version)
```
Comment (bgalek) [inline action.js:38]: remove ;)

File: action.js [hunk 4]
```diff
@@ -1,46 +1,67 @@
...
-    core.info(`next version: v1.0.0${mainBranch ? '' : '-SNAPSHOT'}`);
+    if(type === 'library'){
+      core.info(`next version: v1.0.0${mainBranch ? '' : '-SNAPSHOT'}`);
+      return `v1.0.0${mainBranch ? '' : '-SNAPSHOT'}`;
+    }
 
-    return `v1.0.0${mainBranch ? '' : '-SNAPSHOT'}`;
+    if(type === 'service'){
+      core.info(`next version: v1${mainBranch ? '' : '-SNAPSHOT'}`);
+      return `1${mainBranch ? '' : '-SNAPSHOT'}`;
```
Comment (bgalek) [inline action.js:59]: I would add v1 as a default, tags are string so we should not encourage people to use them like integers

File: incrementer.js [hunk 5]
```diff
@@ -4,12 +4,18 @@ module.exports = {
...
-            const nextVersion = semver.inc(version, step);
-            return prefix + nextVersion + suffix;
+        if ('library' === type.type){
+            if (semver.valid(version)) {
+                const nextVersion = semver.inc(version, step);
+                return prefix + nextVersion + suffix;
+            }
+        }
+        if ('service' === type.type){
+            return prefix + ++version + suffix
```
Comment (bgalek) [inline incrementer.js:18]: template string would be better i think ;) but that's up to you

**Review by bgalek — CHANGES_REQUESTED** [review]

**Comment by bgalek** [comment]
I was talking to Piotr Betkier, and he pointed out an important thing. 
I think we should not imply any convention regards to type of code teams are developing.

Let's change type: service/library to:
- semver 
- custom (your new implementation)
- legacy? (I mean old allegro style with prefix ;))

File: .gitignore (hunk 1 above)
Comment (dkubicki) [reply .gitignore:3]: Set global .gitignore :)

File: action.js (hunk 2 above)
Comment (dkubicki) [reply action.js:3]: yeap, It was my first thought ;)

File: action.js (hunk 4 above)
Comment (dkubicki) [reply action.js:59]: indeed, 58L

**Pull request closed without merging**